version-catalog/api/src/main/kotlin/io/spine/internal/catalog/entry/DependencyEntry.kt

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.catalog.entry
+
+import io.spine.internal.catalog.AlwaysReturnDelegate
+import io.spine.internal.catalog.BundleNotation
+import io.spine.internal.catalog.BundleRecord
+import io.spine.internal.catalog.CatalogRecord
+import io.spine.internal.catalog.LibraryNotation
+import io.spine.internal.catalog.LibraryRecord
+import io.spine.internal.catalog.delegate
+
+/**
+ * A compound catalog entry, which is used to declare complex dependencies.
+ *
+ * Although, many dependencies can be represented by a single [LibraryEntry],
+ * some are a way more complex. They may consist of several modules, versions
+ * and bundles. For example, frameworks.
+ *
+ * This entry can declare a library and version. But, in case it is all you need,
+ * it is better to use a simpler entries: [VersionEntry] and [LibraryEntry].
```

Comment:
What is more complex: "Dependency" or "Library"? To me, that's a tricky question. Essentially a "library" exists on its own. And it may become a "dependency" if it is exposed via some means (such as a Version Catalog), or if it's required by someone in terms of declaring the needs of some Gradle project.

It's going to be very difficult to remember which of three (`Version...`, `Library...`, or `Dependency...`) to pick. This concept needs to be simpler.